devlib/utils/ssh.py

```diff
@@ -460,7 +486,11 @@ def _push_file(cls, sftp, src, dst):
             )
             logger.debug('Trying: {} -> {}'.format(src, new_dst))
             try:
-                sftp.put(src, new_dst)
+                if self.transfer_mgr is not None and self.transfer_mgr.managing.is_set():
+                    with self.transfer_mgr.transfer_in_progress(src):
+                        sftp.put(src, dst)
+                else:
+                    sftp.put(src, dst)
```

Comment:
These `dst`'s should be `new_dst` otherwise this just retires the original command.